Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Miden-node Dockerfile
+
+#### Setup image builder
+FROM rust:1.75.0-bullseye AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y gcc llvm clang bindgen pkg-config
+
+# Setup workdir
+WORKDIR /app
+COPY . miden-node
+RUN cd miden-node && make
```

Comment:
I think it makes sense to use a [`RUN --mount=type=cache`](https://docs.docker.com/reference/dockerfile/#run---mounttypecache) so that we can reuse the host's cargo cache. And just copy the final binary to the image.